src/main/java/com/sda/auction/services/AuctionServiceImpl.java

```diff
@@ -1,14 +1,42 @@
 package com.sda.auction.services;
 
+import com.sda.auction.entities.Auction;
 import com.sda.auction.repositories.AuctionRepository;
 import lombok.AllArgsConstructor;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+import java.util.Optional;
+
 @Service
 @AllArgsConstructor(onConstructor = @__({@Autowired}))
 public class AuctionServiceImpl implements AuctionService {
 
     private final AuctionRepository auctionRepository;
 
+    @Override
+    public List<Auction> findAllAuctions() {
+        return auctionRepository.findAll();
+    }
+
+    @Override
+    public void createAuction(Auction auction) {
+        auctionRepository.save(auction);
+    }
+
+    @Override
+    public Optional<Auction> getAuctionById(Integer id) {
+        return auctionRepository.findById(id);
```

Comment:
Instead of returning the optional, let's instead do a orElseThrows.
We can define our own AuctionNotFoundException, so we will have something like
auctionRepository.findById(id).orElseThrows(()->new AuctionNoutFoundException("Could not find auction"))